lib/pure/json.nim

```diff
@@ -1010,8 +1010,14 @@ when defined(nimFixedForwardGeneric):
       dst = T(jsonNode.num)
 
```

Comment:
```suggestion
macro intValuesSet[T: enum](arg: typedesc[T]): untyped =
  ## Takes an enum and returns a set of integers that contains all
  ## valid enum integer representations of that enum.

  let typedescImpl = getTypeImpl(arg)
  echo typedescImpl[0].lispRepr
  let impl = typedescImpl[1].getImpl[2]
  result = newNimNode(nnkCurly)
  var i = 0
  for idx in 1 ..< impl.len:
    let child = impl[idx]
    case child.kind
    of nnkEnumFieldDef:
      child[1].expectKind nnkIntLit
      result.add child[1]
      i = int(child[1].intVal)
    of nnkIdent:
      result.add newLit(i)
    else:
      child.expectkind({nnkEnumFieldDef, nnkIdent})
    i += 1
```

Reply:
i think you left a debugging print in line 1016?

Reply:
yep I did